### DISCUSSION TITLE
When i select Connection_type_normal energy is produced from the connection rather than flowing from the node.

### DISCUSSION THREAD

**Discussion by SeanByrne08** [spine-tools/SpineOpt.jl#743] [Support]
Hello everyone,

I am trying to use a connection to represent a uni-directional charger. When I select bi-directional it works as it should drawing power from the units that are attached to the node and the energy flows from the node to the connection. When i change the connection type from connection_type_lossless_bidirectional to connection_type_normal energy is produced from the connection rather than flowing from the node. Is there a way to enforce balance at the connection?

Another way to connect the system would be to connect the units "PV" and "grid" to both the "household" and "EV" nodes. However, i have not been able to figure out how to set the capacity of the unit at the unit rather than the at the unit_to_node relationship. Is there a way to set the units capacity and for the energy to be divided between two nodes.

Thank you very much and please ask for any further information required!

![Spine Image](https://github.com/spine-tools/SpineOpt.jl/assets/141043201/9598281e-69df-4fc1-9602-03d74e6709ca)

**Comment by DillonJ** [comment]
Hi @SeanByrne08 

The reason for this is as follows. The `connection_type` parameter is a convenience for creating bi-directional lossless lines. When you use it, you only need to specify the `connection__from_node` and the `connection__to_node`. Each of these relationships, by themselves, only triggers an unconstrained flow from the "from"node to the connection and to the "to" node, from the connection. As you have seen, with these alone, there is no constraint forcing the flows to be equal. When you use `connection_type` = `connection_type_bidirectional_lossess`, under the hood, four additional relationships are created,  `connection__from_node` (from the to node) and `connection__to_node`  (to the "from" node). These allow flows in the reverse direction. Also, two `connection__node__node` relationships (one in each direction). These connection `connection__node__node`  relationships allow us to specify parameters which constrain the from and to flows to each other. In this case, `connection_type` = `connection_type_bidirectional_lossess` creates two fix_ratio_out_in_connection_flow parameters in each direction, both equal to 1 which constrains the flow from the from node to be equal to the flow to the to node and vice-versa.

In short, when you don't use  `connection_type` = `connection_type_bidirectional_lossess`  you need to set the relationships and parameters yourself to achieve your own specific case. In your case, all you are missing is a `connection__node__node` relationship and on that relationhip, a `fix_ratio_out_in_connection_flow` parameter which will constrain the flows as you want. Pay attention that the first node in your `connection__**node**__node`  relationship should be the out node because you are using `fix_ratio_**out**_in_connection_flow`

**Comment by DillonJ** [comment]
A slightly unrelated issue is that in your use case above, I would have used a unit to represent your "EV". This would have avoided the need for the two connections

**Comment by SeanByrne08** [comment]
Hi @DillonJ,

Thank you very much for the quick response! Your model is running correctly after making your changes. I will look at replacing the node "EV" with a unit.
